data/external/scrapers/public_transport.py

```diff
@@ -47,8 +42,59 @@ def scrape_stations():
         for sub in repeat_later:
             if parent := stations.get(sub["parent"]):
                 parent["sub_stations"].append(sub)
-        # remove parent property from sub stations
-        for station in stations.values():
-            for sub in station["sub_stations"]:
-                del sub["parent"]
-        return sorted(stations.values(), key=lambda x: x["lat"])
+        return stations
+
+
+def _load_train_stations(stations: dict) -> list:
+    with open(PUBLIC_TRANSPORT_CACHE_PATH / "train_stations.csv", encoding="utf-8") as file:
+        lines = csv.DictReader(file, delimiter=";")
+        lines = list(filter(lambda x: x["\ufeffHstNummer"], lines))
+        repeat_later = []  # when parent station is not already in dict
+        for line in lines:
+            if (
+                len(line["Globale ID"].split(":")) == 3
+            ):  # parentstation ex. de:09184:460 substation ex. de:09184:460:30:1
+                stations.setdefault(
+                    line["Globale ID"],
+                    {
+                        "station_id": line["Globale ID"],
+                        "name": line["Name ohne Ort"],
+                        "lat": float(line["WGS84 X"].replace(",", ".")),
+                        "lon": float(line["WGS84 Y"].replace(",", ".")),
+                        "sub_stations": [],
+                    },
+                )
+            else:
+                parent_id = ".".join(line["Globale ID"].split(":")[0:3])
+                sub_station = {
+                    "station_id": line["Globale ID"],
+                    "name": line["Name ohne Ort"],
+                    "lat": float(line["WGS84 X"].replace(",", ".")),
+                    "lon": float(line["WGS84 Y"].replace(",", ".")),
+                    "parent": parent_id,
+                }
+
+                if parent := stations.get(parent_id):
+                    parent["sub_stations"].append(sub_station)
+                else:
+                    repeat_later.append(sub_station)
+            for sub in repeat_later:
+                if parent := stations.get(sub["parent"]):
+                    parent["sub_stations"].append(sub)
+        return stations.values()
+
+
+@cached_json("public_transport.json")
+def scrape_stations():
+    """Scrape the stations from the MVV GTFS data and return them as a list of dicts"""
+    _download_file(MVV_GTFS_URL, PUBLIC_TRANSPORT_CACHE_PATH / "fahrplandaten.zip")
+    with ZipFile(PUBLIC_TRANSPORT_CACHE_PATH / "fahrplandaten.zip") as file_zip:
+        file_zip.extract("stops.txt", PUBLIC_TRANSPORT_CACHE_PATH)
+    _download_file(MVV_HST_REPORT_URL, PUBLIC_TRANSPORT_CACHE_PATH / "train_stations.csv")
```

Comment:
I think these files should be downloaded in the functions where they are used given that they are only used there.